brave/brave-core
# Add CountReportLimiter P3A utility, limit counting News card views for P3A

Resolves brave/brave-browser#28356

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

0. Start browser with fresh profile. Enable logging and set logging level to 2. Grep the logs by "CountReportLimiter".
1. Access Brave News, scroll the feed at a normal rate. Ensure "frame event count <= max rate, reporting" is shown in the logs. Ensure the card views metric shows the expected value.
2. Scroll the feed at an excessive rate. Ensure "frame event count > max rate, skipping report" is shown.
3. Scroll at a normal rate. Ensure the log mentioned in step 1 is shown.
